django/forms/utils.py

```diff
@@ -158,10 +158,16 @@ def from_current_timezone(value):
     When time zone support is enabled, convert naive datetimes
     entered in the current time zone to aware datetimes.
     """
+    return_value = value
     if settings.USE_TZ and value is not None and timezone.is_naive(value):
         current_timezone = timezone.get_current_timezone()
         try:
-            return timezone.make_aware(value, current_timezone)
+            return_value = timezone.make_aware(value, current_timezone)
+            if (
+                not timezone._is_pytz_zone(current_timezone) and
+                timezone._datetime_ambiguous_or_imaginary(value, current_timezone)
```

Comment:
`value` or `return_value`? or maybe we should swap these lines:
```python
if (
    not timezone._is_pytz_zone(current_timezone) and
    timezone._datetime_ambiguous_or_imaginary(value, current_timezone)
):
    raise ValueError('Ambiguous or non-existent time.')
```

Reply:
I swapped the lines: you start with an ambiguous naive datetime. Adding the `tz` and `fold` makes it not so. (So `value`)

Reply:
Not 100% sure. I will follow-up with @pganssle about this. 

Testing you can't get ambiguous results regardless of using `value` or `return_value`, and adjustments of fold: 

```
>>> import datetime
>>> from django.utils import timezone
>>> import zoneinfo
>>> d = datetime.datetime(2015, 3, 29, 2, 30)
>>> d.fold
0
>>> PARIS_ZI = zoneinfo.ZoneInfo('Europe/Paris')
>>> timezone._datetime_ambiguous_or_imaginary(d, PARIS_ZI)
True
>>> a = d.replace(tzinfo=PARIS_ZI)
>>> timezone._datetime_ambiguous_or_imaginary(a, PARIS_ZI)
True
>>> a2 = a.replace(fold=1)
>>> timezone._datetime_ambiguous_or_imaginary(a2, PARIS_ZI)
True
```

Reply:
Yeah, this is fine. I think the order-swap was to avoid unnecessary work in the common case where the user is using `pytz`, and because frequently people prefer to avoid multiple returns.

If you're OK with multiple returns (I don't think it's really any worse in this case), I think that `_is_pytz_zone` is fairly lightweight — I would guess it's mostly a bit of function call overhead, so this swapped approach seems fine.